domain/network/network.py

```diff
@@ -0,0 +1,171 @@
+    :type state: :py:class:`domain.network.network.NodeState`
+
+    :param last_message: Time passed since the last message that node received in seconds
+    :type last_message: float
+
+    :param latest_timestamp: Timestamp of the latest block of the node
+    :type latest_timestamp: int
+
+    :param sync_timestamp: Timestamp of the last synchronized block of the node
+    :type sync_timestamp: int
+
+    :param warning_flags: List of node warnings, if any
+    :type warning_flags: List[str]
+
+    """
+    id: str
+    app_version: str
+    uptime: float
+    address: str
+    state: NodeState
+    last_message: float
+    latest_timestamp: int
+    sync_timestamp: int
+    warning_flags: List[str]
+
+
+@dataclass
+class Network:
+    """ Network information of a given node
+
+    :param id: Node id hash
+    :type id: str
+
+    :param app_version: The current version of app running on node
+    :type app_version: str
+
+    :param state: Current state of the node
+    :type state: :py:class:`domain.network.network.NodeState`
+
+    :param network: Current network. Can be `mainnet`, any version of `testnet` or other
+    :type network: str
+
+    :param uptime: Time of node activity in seconds
+    :type uptime: float
+
+    :param first_timestamp: Timestamp of the first block of the node
+    :type first_timestamp: int
+
+    :param latest_timestamp: Timestamp of the latest block of the node
+    :type latest_timestamp: int
+
+    :param entrypoints: List of node entrypoints
+    :type entrypoints: List[str]
+
+    :param known_peers: List of ids of peers known by the node
+    :type known_peers: List[str]
+
+    :param connected_peers: List of peers connected with the node
+    :type connected_peers: List[Peer]
+
+    """
+    id: str
+    app_version: str
+    state: NodeState
+    network: str
+    uptime: float
+    first_timestamp: int
+    latest_timestamp: int
+    entrypoints: List[str]
+    known_peers: List[str]
+    connected_peers: List[Peer]
+
+    def to_dict(self) -> dict:
+        """ Convert a Network instance into dict
+
+        :return: The dict representation of the Network
+        :rtype: dict
+        """
+        return asdict(self)
+
+    @classmethod
+    def from_dict(cls, dikt: dict) -> 'Network':
+        """ Creates a new Network instance from a given dict (inverse operation of `to_dict`)
+
+        :param dikt: Dict with Network structure and data
+        :type dikt: dict
+
+        :return: The new instance
+        :rtype: :py:class:`domain.network.network.Network`
+        """
+        dikt['state'] = NodeState(dikt['state'])
+
+        connected_peers = []
+
+        for peer in dikt['connected_peers']:
+            peer['state'] = NodeState(peer['state'])
+            connected_peers.append(Peer(**peer))
+
+        dikt['connected_peers'] = connected_peers
+
+        return Network(**dikt)
+
+    @classmethod
+    def from_status_dict(cls, status: dict) -> 'Network':
```

Comment:
I think we could also have a class representing the data we receive from the APIs, like a data-transfer object (DTO), maybe. In this case, we would have a class for this `status` object.

It would be turned into a object as soon as we receive data from the API (which happens in `usecases/collect_nodes_statuses.py`). With a lib like [dacite](https://pypi.org/project/dacite/), it would be as simple as this to transform the data into a object:
```
from_dict(data_class=StatusDTO, data=status_dict)
```

This pattern avoids that other parts of the code have to infere the structure that is returned from an API call, which is what is happening in the code below. Everything will be typed and less error prone.

Reply:
I agree! I've already created an issue for it: https://github.com/HathorNetwork/hathor-explorer-service/issues/4. 

Didn't knew dacite. Liked a lot, i'll add it,